subsystems/grippersubsystem.py

```diff
@@ -0,0 +1,97 @@
+# each roller alternates meaning if you were to pick up a cube you would eject a cone
+from enum import Enum, auto
+from commands2 import SubsystemBase
+from rev import SparkMaxLimitSwitch
+from wpilib import SmartDashboard
+
+from util.simneo import NEOBrushless
+
+import constants
+
+
+class GripperSubsystem(SubsystemBase):
+    class GripperState(Enum):
+        CubeGrabForward = auto()  # Cube gripper wheels, motor will move right
+        ConeGrabBackwards = auto()  # Cone gripper wheels, motor will move left
+        HoldingState = (
+            auto()
+        )  # holds the cargo in the gripper, no movement in the motor
+
+    def __init__(self) -> None:
+        SubsystemBase.__init__(self)
+        self.setName(__class__.__name__)
+
+        self.motorCubeCone = NEOBrushless(
+            constants.kConeCubeCANID,
+            constants.kConeCubePIDSlot,
+            constants.kConeCubePGain,
+            constants.kConeCubeIGain,
+            constants.kConeCubeDGain,
+            enableLimitSwitches=False,
+            limitSwitchPolarity=SparkMaxLimitSwitch.Type.kNormallyClosed,
+        )
+        self.motorCubeCone.setSmartCurrentLimit(limit=constants.kIntakeMotorVoltage)
+
+        self.state = GripperSubsystem.GripperState.HoldingState
+        self.cubeSensor = lambda: self.motorCubeCone.getLimitSwitch(
+            NEOBrushless.LimitSwitch.Forwards
+        )
+        self.coneSensor = lambda: self.motorCubeCone.getLimitSwitch(
+            NEOBrushless.LimitSwitch.Backwards
+        )
+        SmartDashboard.putBoolean(constants.kCubeMode, False)
+
+    def periodic(self) -> None:
+        SmartDashboard.putString(constants.kIntakeState, str(self.state))
+        SmartDashboard.putNumber(
+            constants.kIntakeMotorRPMKey,
+            self.motorCubeCone.get(NEOBrushless.ControlMode.Velocity)
+            / constants.kIntakeGearRatio,
+        )
+        SmartDashboard.putBoolean(constants.kCubeLoaded, self.cubeSensor())
+        SmartDashboard.putBoolean(constants.kConeLoaded, self.coneSensor())
+        if self.state == self.GripperState.CubeGrabForward:
```

Comment:
since we are currently now thinking of it as intake and outtake, it would not make sense to consider the states to be "cube grab" and "cone grab" but rather an intake and outtake that changes depending on whether we are in code or cube mode

lots of variables to rename to reflect this change